src/components/App.jsx

```diff
@@ -342,20 +343,26 @@ class App extends Component {
                     <h2>No Registration. No Fees.</h2>
                   </div>
                 </div>
-                <Widget isConnected={this.state.network.isConnected}
-                        section={this.state.section}
-                        network={this.state.network.network}
-                        account={this.state.network.defaultAccount}
-                        proxy={this.state.proxy}
-                        trade={this.state.trade}
-                        balances={this.state.balances}
-                        showTxMessage={this.state.showTxMessage}
-                        transactions={this.state.transactions}
-                        setMainState={this.setMainState}
-                        fasterGasPrice={this.fasterGasPrice}
-                        doTrade={this.doTrade}
-                        reset={this.reset}
-                        getProxy={this.getProxy} />
+                <WidgetContext.Provider value={
```

Comment:
You can pass `value ={state:this.state}`

Also I think there will be separation for sure